tailscale/client.go

```diff
@@ -45,6 +47,25 @@ type (
 const baseURL = "https://api.tailscale.com"
 const contentType = "application/json"
 
+// NewClientFromOAuth returns a new instance of the Client type that will perform operations against a chosen tailnet using
+// OAuth Credentials. The Client will generate a new API key against the Tailscale OAuth token endpoint.
+// If the API key expires, it will not be automatically renewed by the Client.
+// Additional options can be provided, see ClientOption for more details.
+func NewClientFromOAuth(oauthClientID, oauthClientSecret, tailnet string, options ...ClientOption) (*Client, error) {
+	var oauthConfig = &clientcredentials.Config{
```

Comment:
Shouldn't `scopes` be specified as well?

Reply:
Tailscale OAuth clients already have a set of scopes configured on them, and by default access tokens will have those same scopes.  You only need to specify scopes here if you want to further limit the scopes applied, but I'm not sure if that will be very common.

Reply:
My latest push adds support for passing in scopes.